cpu/atxmega/periph/cpuid.c

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2021 Gerson Fernando Budke <nandojve@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_atxmega
+ * @ingroup     cpu_atxmega_periph
+ * @{
+ *
+ * @file
+ * @brief       Low-level CPUID driver implementation
+ *
+ * @author      Gerson Fernando Budke <nandojve@gmail.com>
+ *
+ * @}
+ */
+
+#include "periph_cpu.h"
+#include "cpu_nvm.h"
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+void cpuid_get(void *id)
+{
+    uint8_t *addr = id;
+
+    addr[0x0] = nvm_read_production_signature_row(
+        nvm_get_production_signature_row_offset(LOTNUM0));
+    addr[0x1] = nvm_read_production_signature_row(
+        nvm_get_production_signature_row_offset(LOTNUM1));
+    addr[0x2] = nvm_read_production_signature_row(
+        nvm_get_production_signature_row_offset(LOTNUM2));
+    addr[0x3] = nvm_read_production_signature_row(
+        nvm_get_production_signature_row_offset(LOTNUM3));
+    addr[0x4] = nvm_read_production_signature_row(
+        nvm_get_production_signature_row_offset(LOTNUM4));
+    addr[0x5] = nvm_read_production_signature_row(
+        nvm_get_production_signature_row_offset(LOTNUM5));
+
+    addr[0x6] = nvm_read_production_signature_row(
+        nvm_get_production_signature_row_offset(WAFNUM));
+
+    addr[0x7] = nvm_read_production_signature_row(
+        nvm_get_production_signature_row_offset(COORDX0));
+    addr[0x8] = nvm_read_production_signature_row(
+        nvm_get_production_signature_row_offset(COORDX1));
+    addr[0x9] = nvm_read_production_signature_row(
+        nvm_get_production_signature_row_offset(COORDY0));
+    addr[0xa] = nvm_read_production_signature_row(
+        nvm_get_production_signature_row_offset(COORDY1));
+
+#if ENABLE_DEBUG
+    DEBUG("CPUID: ");
+    for (uint8_t i = 0; i < CPUID_LEN; i++) {
+        DEBUG("%02x ", addr[i]);
+    }
+    DEBUG("\n");
+#endif
```

Comment:
```suggestion
    if (IS_ACTIVE(ENABLE_DEBUG)) {
        DEBUG("CPUID: ");
        for (uint8_t i = 0; i < CPUID_LEN; i++) {
            DEBUG("%02x ", addr[i]);
        }
        DEBUG("\n");
    }
```

You could rely on the C compiler to eliminate dead branches when the condition in the `if` is a compile time constant. The `IS_ACTIVE()` is a preprocessor macro that does the following:

1. If `arg` in `IS_ACTIVE(arg)` is a defined preprocessor macro, it will evaluate to `arg`
2. If `arg` is undefined, it will evaluate to `0`

Since `ENABLE_DEBUG` is always defined here, there is little reason for using it here. But it is more consistent with other code, where preprocessor macros are only defined when they are "active".